credential/template/options.go

```diff
@@ -0,0 +1,100 @@
+package template
+
+import (
+	"time"
+
+	"github.com/axone-protocol/axone-sdk/credential"
+)
+
+type Option[T credential.Descriptor] func(descriptor T)
+
+type HasID interface {
+	setID(string)
+}
+
+func WithID[T interface {
+	HasID
+	credential.Descriptor
+}](id string) Option[T] {
+	return func(descriptor T) {
+		descriptor.setID(id)
+	}
+}
```

Comment:
Awesome 😍

Reply:
@amimart The inconvenience is that the compiler cannot infer `T` when initializing options on each `Descriptor` because it needs to know the specific type for the options. 
This results in code with a lot of [syntactic salt](https://fr.wikipedia.org/wiki/Sucre_syntaxique#Sel_syntaxique) (by opposition of syntactic sugar like we talk last week, you were right 😛) : 

```go
func NewDataset(datasetDID, title string, opts ...Option[*DatasetDescriptor]) *DatasetDescriptor {...}
```

```go
NewDataset(
    "datasetID",
    "title",
    WithID[*DatasetDescriptor]("id"),
    WithDescription[*DatasetDescriptor]("description"),
    WithFormat[*DatasetDescriptor]("format"),
    WithTags[*DatasetDescriptor]([]string{"tag1", "tag2"}),
    WithTopic[*DatasetDescriptor]("topic")
)
```